src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FixLegacyResourceDesignerStep.cs

```diff
@@ -0,0 +1,158 @@
+				return;
+			}
+			designerType = designerAssembly.MainModule.GetTypes ().FirstOrDefault (x => x.FullName == $"{DesignerAssemblyName}.Resource");
+			if (designerType == null) {
+				LogMessage ($"   Did not find {DesignerAssemblyName}.Resource type. It was probably linked out.");
+				return;
+			}
+			lookup = BuildResourceDesignerPropertyLookup (designerType);
+			return;
+		}
+		internal override bool ProcessAssemblyDesigner (AssemblyDefinition assembly)
+		{
+			if (designerAssembly == null || designerType == null) {
+				LogMessage ($"   Not using {DesignerAssemblyName}");
+				return false;
+			}
+
+			if (!FindResourceDesigner (assembly, mainApplication: false, out TypeDefinition designer, out CustomAttribute designerAttribute)) {
+				LogMessage ($"   {assembly.Name.Name} has no designer. ");
+				return false;
+			}
+
+			LogMessage ($"   {assembly.Name.Name} has a designer. ");
+			LogMessage ($"   BaseType: {designer.BaseType.FullName}. ");
+			if (designer.BaseType.FullName == $"{DesignerAssemblyName}.Resource") {
+				LogMessage ($"   {assembly.Name.Name} has already been processed. ");
+				return false;
+			}
+
+			LogMessage ($"    Adding reference {designerAssembly.Name.Name}.");
+			assembly.MainModule.AssemblyReferences.Add (designerAssembly.Name);
+			var importedDesignerType = assembly.MainModule.ImportReference (designerType.Resolve ());
+
+			LogMessage ($"    FixupAssemblyTypes {assembly.Name.Name}.");
+			// now replace all ldsfld with a call to the property get_ method.
+			FixupAssemblyTypes (assembly, designer);
+
+			LogMessage ($"    ClearDesignerClass {assembly.Name.Name}.");
+			// then clean out the designer.
+			ClearDesignerClass (designer, completely: true);
+			designer.BaseType = importedDesignerType;
+			return true;
+		}
+
+		Dictionary<string, MethodDefinition> BuildResourceDesignerPropertyLookup (TypeDefinition type)
+		{
+			LogMessage ($"     Building Designer Lookups for {type.FullName}");
+			var output = new Dictionary<string, MethodDefinition> (StringComparer.OrdinalIgnoreCase);
+			foreach (TypeDefinition definition in type.NestedTypes)
+			{
+				foreach (PropertyDefinition property in definition.Properties)
+				{
+					string key = $"{definition.Name}::{property.Name}";
+					if (!output.ContainsKey (key)) {
+						LogMessage ($"          Adding {key}");
+						output.Add(key, property.GetMethod);
+					}
+				}
+			}
+			return output;
+		}
+
+		protected override void FixBody (MethodBody body, TypeDefinition designer)
+		{
+			// replace
+			// IL_0068: ldsfld int32 Xamarin.Forms.Platform.Android.Resource/Layout::Toolbar
+			// with
+			// call int32 Xamarin.Forms.Platform.Android.Resource/Layout::get_Toolbar()
+			string designerFullName = $"{designer.FullName}/";
+			var processor = body.GetILProcessor ();
+			Dictionary<Instruction, Instruction> instructions = new Dictionary<Instruction, Instruction>();
+			foreach (var i in body.Instructions)
+			{
+				string line = i.ToString ();
```

Comment:
I feel like it would be "better" to assert that [`i.OpCode`](https://github.com/jbevain/cecil/blob/4ad9c0f0174347763165caf724f4f13a554450d3/Mono.Cecil.Cil/Instruction.cs#L311) is [`OpCodes.Ldsfld`](https://github.com/jbevain/cecil/blob/4ad9c0f0174347763165caf724f4f13a554450d3/Mono.Cecil.Cil/OpCodes.cs#L510) and that [`i.Operand`](https://github.com/jbevain/cecil/blob/4ad9c0f0174347763165caf724f4f13a554450d3/Mono.Cecil.Cil/Instruction.cs#L36)`.ToString()` contains `designerFullName`.

Reply:
Great feedback, I can do the OpCode check just before the `i.ToString()` and `continue`. It should speed things up a bit 👍